In [1]:
import pandas as pd
import numpy as np

## Loading the dataset

In [6]:
# Reading datasets by using read_csv from pandas package
# Reading datasets by using read_csv from pandas package
movie = pd.read_csv(r"C:\Users\New\Desktop\Jupyter Notebook\Data Science Project\movie.csv")
ratings = pd.read_csv(r"C:\Users\New\Desktop\Jupyter Notebook\Data Science Project\ratings.csv")
user = pd.read_csv(r"C:\Users\New\Desktop\Jupyter Notebook\Data Science Project\user.csv")


## Data Overview

### *User Dataset*

In [7]:
# Displaying the User Dataset
user.head()

,user id,age,gender,occupation,zip code
0,1,24,M,technician,85711
1,2,53,F,other,94043
2,3,23,M,writer,32067
3,4,24,M,technician,43537
4,5,33,F,other,15213


In [8]:
user.shape

(943, 5)

In [9]:
user.info()

<class 'pandas.DataFrame'>
RangeIndex: 943 entries, 0 to 942
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   user id     943 non-null    int64
 1   age         943 non-null    int64
 2   gender      943 non-null    str  
 3   occupation  943 non-null    str  
 4   zip code    943 non-null    str  
dtypes: int64(2), str(3)
memory usage: 37.0 KB


In [10]:
user.describe()

,user id,age
count,943.000000,943.000000
mean,472.000000,34.051962
std,272.364951,12.192740
min,1.000000,7.000000
25%,236.500000,25.000000
50%,472.000000,31.000000
75%,707.500000,43.000000
max,943.000000,73.000000


- *We can see from the `age` column that the average age of users who have rated the movies is 34 years*
- *The age of users ranges from 7 years (min or minimum) to 73 years (max or maximum)*
- *Since user id is an unique identifier of the users who have rated the movies, interpreting its summary statistics won't yield meaningful insights.*

### *Movie Dataset*

In [12]:
#displaying the movie dataset
movie.head()

,movie id,movie title,release date,Action,Adventure,Animation,Childrens,Comedy,Crime,Documentary,...,Fantasy,Film-Noir,Horror,Musical,Mystery,Romance,Sci-Fi,Thriller,War,Western
0,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,0,0,0,0
1,2,GoldenEye,1-Jan-95,1,1,0,0,0,0,0,...,0,0,0,0,0,0,0,1,0,0
2,3,Four Rooms,1-Jan-95,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,1,0,0
3,4,Get Shorty,1-Jan-95,1,0,0,0,1,0,0,...,0,0,0,0,0,0,0,0,0,0
4,5,Copycat,1-Jan-95,0,0,0,0,0,1,0,...,0,0,0,0,0,0,0,1,0,0


In [13]:
#Understanding the shape of the movie dataset
movie.shape

(1680, 21)

In [14]:
# Checking the data types of the columns for the movie dataset
movie.info()

<class 'pandas.DataFrame'>
RangeIndex: 1680 entries, 0 to 1679
Data columns (total 21 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   movie id      1680 non-null   int64
 1   movie title   1680 non-null   str  
 2   release date  1680 non-null   str  
 3   Action        1680 non-null   int64
 4   Adventure     1680 non-null   int64
 5   Animation     1680 non-null   int64
 6   Childrens     1680 non-null   int64
 7   Comedy        1680 non-null   int64
 8   Crime         1680 non-null   int64
 9   Documentary   1680 non-null   int64
 10  Drama         1680 non-null   int64
 11  Fantasy       1680 non-null   int64
 12  Film-Noir     1680 non-null   int64
 13  Horror        1680 non-null   int64
 14  Musical       1680 non-null   int64
 15  Mystery       1680 non-null   int64
 16  Romance       1680 non-null   int64
 17  Sci-Fi        1680 non-null   int64
 18  Thriller      1680 non-null   int64
 19  War           1680 non-null   int64
 2

In [15]:
movie.describe()

,movie id,Action,Adventure,Animation,Childrens,Comedy,Crime,Documentary,Drama,Fantasy,Film-Noir,Horror,Musical,Mystery,Romance,Sci-Fi,Thriller,War,Western
count,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000,1680.000000
mean,841.525595,0.149405,0.080357,0.025000,0.072619,0.300595,0.064881,0.029762,0.431548,0.013095,0.014286,0.054762,0.033333,0.036310,0.147024,0.060119,0.149405,0.042262,0.016071
std,485.609591,0.356593,0.271926,0.156171,0.259587,0.458653,0.246389,0.169980,0.495440,0.113717,0.118701,0.227583,0.179559,0.187115,0.354235,0.237778,0.356593,0.201246,0.125788
min,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,421.750000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,841.500000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
75%,1261.250000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
max,1682.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000


### *Ratings Dataset*

In [16]:
ratings.head()


,user id,movie id,rating,timestamp
0,196,242,3,881250949
1,186,302,3,891717742
2,22,377,1,878887116
3,244,51,2,880606923
4,166,346,1,886397596


In [17]:
#Understanding the shape of the ratings dataset
ratings.shape


(100000, 4)

In [18]:
ratings.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 4 columns):
 #   Column     Non-Null Count   Dtype
---  ------     --------------   -----
 0   user id    100000 non-null  int64
 1   movie id   100000 non-null  int64
 2   rating     100000 non-null  int64
 3   timestamp  100000 non-null  int64
dtypes: int64(4)
memory usage: 3.1 MB


In [19]:
# Getting the statistical summary for the ratings dataset
ratings.describe()

,user id,movie id,rating,timestamp
count,100000.00000,100000.000000,100000.000000,1.000000e+05
mean,462.48475,425.530130,3.529860,8.835289e+08
std,266.61442,330.798356,1.125674,5.343856e+06
min,1.00000,1.000000,1.000000,8.747247e+08
25%,254.00000,175.000000,3.000000,8.794487e+08
50%,447.00000,322.000000,4.000000,8.828269e+08
75%,682.00000,631.000000,4.000000,8.882600e+08
max,943.00000,1682.000000,5.000000,8.932866e+08


## Finding How many Movies belong to a particular genre

In [20]:
movie.head()

,movie id,movie title,release date,Action,Adventure,Animation,Childrens,Comedy,Crime,Documentary,...,Fantasy,Film-Noir,Horror,Musical,Mystery,Romance,Sci-Fi,Thriller,War,Western
0,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,0,0,0,0
1,2,GoldenEye,1-Jan-95,1,1,0,0,0,0,0,...,0,0,0,0,0,0,0,1,0,0
2,3,Four Rooms,1-Jan-95,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,1,0,0
3,4,Get Shorty,1-Jan-95,1,0,0,0,1,0,0,...,0,0,0,0,0,0,0,0,0,0
4,5,Copycat,1-Jan-95,0,0,0,0,0,1,0,...,0,0,0,0,0,0,0,1,0,0


## *Analyzing the dataset using pandas*

In [22]:
#Getting all the column of the Movie dataset
movie.columns

Index(['movie id', 'movie title', 'release date', 'Action', 'Adventure',
       'Animation', 'Childrens', 'Comedy', 'Crime', 'Documentary', 'Drama',
       'Fantasy', 'Film-Noir', 'Horror', 'Musical', 'Mystery', 'Romance',
       'Sci-Fi', 'Thriller', 'War', 'Western'],
      dtype='str')

In [23]:
genres = ['Action', 'Adventure',
       'Animation', 'Childrens', 'Comedy', 'Crime', 'Documentary', 'Drama',
       'Fantasy', 'Film-Noir', 'Horror', 'Musical', 'Mystery', 'Romance',
       'Sci-Fi', 'Thriller', 'War', 'Western']

In [24]:
# using the len() function to calculate the number of genres
print('The number of genres present in the movie dataset are ', len(genres))

The number of genres present in the movie dataset are  18


In [25]:
    # printing the genres list
genres

['Action',
 'Adventure',
 'Animation',
 'Childrens',
 'Comedy',
 'Crime',
 'Documentary',
 'Drama',
 'Fantasy',
 'Film-Noir',
 'Horror',
 'Musical',
 'Mystery',
 'Romance',
 'Sci-Fi',
 'Thriller',
 'War',
 'Western']

In [26]:
movie[genres].sum()

Action         251
Adventure      135
Animation       42
Childrens      122
Comedy         505
Crime          109
Documentary     50
Drama          725
Fantasy         22
Film-Noir       24
Horror          92
Musical         56
Mystery         61
Romance        247
Sci-Fi         101
Thriller       251
War             71
Western         27
dtype: int64

In [27]:
#Sorting according to ascending order.
movie[genres].sum().sort_values(ascending = False) #ascending= False will sort the values in descending order. By default the parameter ascending is set to True

Drama          725
Comedy         505
Thriller       251
Action         251
Romance        247
Adventure      135
Childrens      122
Crime          109
Sci-Fi         101
Horror          92
War             71
Mystery         61
Musical         56
Documentary     50
Animation       42
Western         27
Film-Noir       24
Fantasy         22
dtype: int64

*Business Insight*

- *Movies are distributed across 18 genres in total*
- *We can observe that 'Drama' genre has the highest number of movies with 725 in total followed by 'Comedy' at 505*
- *The top five genres with the highest number of movies are 'Drama', 'Comedy', 'Action', 'Thriller', and 'Romance'*
- *'Fantasy' genre has the lowest number of movies (22 in count)*

## Finding the movies with more than one genre

In [28]:
#adding a column to the movie dataset for the number of genre each movie have
movie["Number of Genres"] = movie.loc[:, genres].sum(axis=1)


In [29]:
movie.head()

,movie id,movie title,release date,Action,Adventure,Animation,Childrens,Comedy,Crime,Documentary,...,Film-Noir,Horror,Musical,Mystery,Romance,Sci-Fi,Thriller,War,Western,Number of Genres
0,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,0,0,0,3
1,2,GoldenEye,1-Jan-95,1,1,0,0,0,0,0,...,0,0,0,0,0,0,1,0,0,3
2,3,Four Rooms,1-Jan-95,0,0,0,0,0,0,0,...,0,0,0,0,0,0,1,0,0,1
3,4,Get Shorty,1-Jan-95,1,0,0,0,1,0,0,...,0,0,0,0,0,0,0,0,0,3
4,5,Copycat,1-Jan-95,0,0,0,0,0,1,0,...,0,0,0,0,0,0,1,0,0,3


In [30]:
#this will subset the data as per the given condition
movie[movie['Number of Genres']>1]

,movie id,movie title,release date,Action,Adventure,Animation,Childrens,Comedy,Crime,Documentary,...,Film-Noir,Horror,Musical,Mystery,Romance,Sci-Fi,Thriller,War,Western,Number of Genres
0,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,0,0,0,3
1,2,GoldenEye,1-Jan-95,1,1,0,0,0,0,0,...,0,0,0,0,0,0,1,0,0,3
3,4,Get Shorty,1-Jan-95,1,0,0,0,1,0,0,...,0,0,0,0,0,0,0,0,0,3
4,5,Copycat,1-Jan-95,0,0,0,0,0,1,0,...,0,0,0,0,0,0,1,0,0,3
6,7,Twelve Monkeys,1-Jan-95,0,0,0,0,0,0,0,...,0,0,0,0,0,1,0,0,0,2
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1666,1669,MURDER and murder,20-Jun-97,0,0,0,0,0,1,0,...,0,0,0,1,0,0,0,0,0,3
1667,1670,Tainted,1-Feb-98,0,0,0,0,1,0,0,...,0,0,0,0,0,0,1,0,0,2
1670,1673,Mirage,1-Jan-95,1,0,0,0,0,0,0,...,0,0,0,0,0,0,1,0,0,2
1676,1679,B. Monkey,6-Feb-98,0,0,0,0,0,0,0,...,0,0,0,0,1,0,1,0,0,2


In [31]:
#Getting the shape
movie[movie['Number of Genres']>1].shape

(849, 22)

\- *There are 849 movies that are having more than one genre.*

In [32]:
movie.shape

(1680, 22)

- *The original movie dataset contained 1680 movies. So out of 1680 movies, 849 i.e around 50% of the movies are having more than one genre.*



### Business Insight

- *Out of 1680 movies, 849 movies comprising around 50% are having more than one genre and 831 movies have only single genre*
- *This shows that we have an almost equal ratio (50:50) of movies having more than one genre and having single genre*

In [33]:
df_merge = movie.merge(ratings, on = 'movie id', how = 'inner')
df_merge.head()

,movie id,movie title,release date,Action,Adventure,Animation,Childrens,Comedy,Crime,Documentary,...,Mystery,Romance,Sci-Fi,Thriller,War,Western,Number of Genres,user id,rating,timestamp
0,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,308,4,887736532
1,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,287,5,875334088
2,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,148,4,877019411
3,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,280,4,891700426
4,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,66,3,883601324


In [ ]:
#To find the average rating for each genre, we can use a for loop to iterate through the genres list and calculate the mean rating for each genre using the df_merge dataframe. The code below will print the average rating for each genre.
for i in genres:
    print(i,':' , df_merge[df_merge[i]==1].rating.mean())

Action : 3.480245417953027
Adventure : 3.503526503308369
Animation : 3.5766990291262135
Childrens : 3.3532442216652742
Comedy : 3.3940734781442745
Crime : 3.6322780881440098
Documentary : 3.6728232189973613
Drama : 3.6873793708484772
Fantasy : 3.2152366863905324
Film-Noir : 3.9215233698788228
Horror : 3.2903893172841827
Musical : 3.521396851029471
Mystery : 3.63813155386082
Romance : 3.621704948358255
Sci-Fi : 3.5607227022780834
Thriller : 3.5090069495245064
War : 3.815811874866993
Western : 3.6132686084142396


- *'Film-Noir' genre is having the highest average ratings with an average rating of 3.92 followed by 'War' genre with rating of 3.81*
- *'Fantasy' genre is having the lowest average ratings with an average rating of 3.21*
- *Among 18 genres, only 5 genres have received a rating below 3.5 which means around 72% of genres are rated above average*

### Finding the Movies most perferred by user

In [35]:
df_merge.head()

,movie id,movie title,release date,Action,Adventure,Animation,Childrens,Comedy,Crime,Documentary,...,Mystery,Romance,Sci-Fi,Thriller,War,Western,Number of Genres,user id,rating,timestamp
0,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,308,4,887736532
1,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,287,5,875334088
2,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,148,4,877019411
3,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,280,4,891700426
4,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,0,0,0,0,3,66,3,883601324


In [36]:
avg_rating = df_merge.groupby(['movie title'])[['rating']].mean().reset_index() # reset_index will reset the index of dataframe to default indexing (0 to number of rows minus 1)
avg_rating

,movie title,rating
0,'Til There Was You,2.333333
1,1-900,2.600000
2,101 Dalmatians,2.908257
3,12 Angry Men,4.344000
4,187,3.024390
...,...,...
1652,Young Guns,3.207921
1653,Young Guns II,2.772727
1654,"Young Poisoner's Handbook, The",3.341463
1655,Zeus and Roxanne,2.166667


In [ ]:
#replacing the column name 'rating' with 'Avg_rating' in the avg_rating dataframe
avg_rating.rename(columns={'rating':'Avg_rating'},inplace=True)
avg_rating.head()

,movie title,Avg_rating
0,'Til There Was You,2.333333
1,1-900,2.600000
2,101 Dalmatians,2.908257
3,12 Angry Men,4.344000
4,187,3.024390


In [ ]:
#checking the movie title with the highest average rating
avg_rating.sort_values(ascending=False, by= 'Avg_rating')

,movie title,Avg_rating
30,Aiqing wansui,5.0
1170,Prefontaine,5.0
631,"Great Day in Harlem, A",5.0
1354,Someone Else's America,5.0
1382,Star Kid,5.0
...,...,...
482,"Eye of Vichy, The (Oeil de Vichy, L')",1.0
1604,Wend Kuuni (God's Gift),1.0
1648,Yankee Zulu,1.0
1640,"Woman in Question, The",1.0


In [39]:
avg_rating[avg_rating['Avg_rating']==5] #it will subset the data as per the specified condition

,movie title,Avg_rating
30,Aiqing wansui,5.0
461,Entertaining Angels: The Dorothy Day Story,5.0
631,"Great Day in Harlem, A",5.0
942,Marlene Dietrich: Shadow and Light (,5.0
1170,Prefontaine,5.0
1270,"Saint of Fort Washington, The",5.0
1274,Santa with Muscles,5.0
1354,Someone Else's America,5.0
1382,Star Kid,5.0
1466,They Made Me a Criminal,5.0


- *We can observe movies like **Great Day in Harlem, A, Prefontaine** etc are top rated movies*
- *Movies like **Shadow of Angels (Schatten der Engel), Power 98**, etc are least rated*
- *There are 10 movies which have been rated as 5.0*

Some movies are rated more than once by different users, and hence it may affect the overall rating of the movie.

### Finding movies rated most of the time

In [60]:
movie_count = df_merge.groupby(['movie title'])['rating'].count().reset_index()
movie_count

,movie title,rating
0,'Til There Was You,9
1,1-900,5
2,101 Dalmatians,109
3,12 Angry Men,125
4,187,41
...,...,...
1652,Young Guns,101
1653,Young Guns II,44
1654,"Young Poisoner's Handbook, The",41
1655,Zeus and Roxanne,6


In [61]:
movie_count.rename(columns={'rating':'Rating_counts'},inplace=True)
movie_count.head()

,movie title,Rating_counts
0,'Til There Was You,9
1,1-900,5
2,101 Dalmatians,109
3,12 Angry Men,125
4,187,41


In [62]:
movie_count.sort_values(ascending=False, by= 'Rating_counts')

,movie title,Rating_counts
1393,Star Wars,583
332,Contact,509
497,Fargo,508
1232,Return of the Jedi,507
858,Liar Liar,485
...,...,...
1576,"Very Natural Thing, A",1
1587,"Walk in the Sun, A",1
1088,Office Killer,1
1123,Paris Was a Woman,1


- **Star Wars** has been rated most number of times (**583 times**) and there are few movies which have least rating counts like **Paris Was a Woman	,Á köldum klaka (Cold Fever)**.*

In [63]:
## Checking for the movies which have more than 100 ratings. This will help us to find the movies which are rated by more than 100 users.
movie_100 = movie_count[movie_count['Rating_counts']>100] #this will subset the dataset 'movie_count' as per the specified condition
movie_100


,movie title,Rating_counts
2,101 Dalmatians,109
3,12 Angry Men,125
7,2001: A Space Odyssey,259
15,Absolute Power,127
16,"Abyss, The",151
...,...,...
1627,Willy Wonka and the Chocolate Factory,326
1638,"Wizard of Oz, The",246
1646,"Wrong Trousers, The",118
1651,Young Frankenstein,200


*There are 334 movies which are rated more than 100 times by the users*

In [64]:
## Checking for the movies which have less than 100 ratings. This will help us to find the movies which are rated by less than 100 users.
movie_1001 = movie_count[movie_count['Rating_counts']<100] #this will subset the dataset 'movie_count' as per the specified condition
movie_1001


,movie title,Rating_counts
0,'Til There Was You,9
1,1-900,5
4,187,41
5,2 Days in the Valley,93
6,"20,000 Leagues Under the Sea",72
...,...,...
1650,You So Crazy,1
1653,Young Guns II,44
1654,"Young Poisoner's Handbook, The",41
1655,Zeus and Roxanne,6


In [65]:
#Finding movies top rated by users. This will help us to find the movies which are rated by more than 100 users.
avg_rating.head()

,movie title,Avg_rating
0,'Til There Was You,2.333333
1,1-900,2.600000
2,101 Dalmatians,2.908257
3,12 Angry Men,4.344000
4,187,3.024390


In [66]:
movie_100.head()

,movie title,Rating_counts
2,101 Dalmatians,109
3,12 Angry Men,125
7,2001: A Space Odyssey,259
15,Absolute Power,127
16,"Abyss, The",151


In [67]:
#Merging the two dataframes avg_rating and movie_100 on the column 'movie title' using inner join.
df_top = avg_rating.merge(movie_100, on = 'movie title', how = 'inner')
df_top

,movie title,Avg_rating,Rating_counts
0,101 Dalmatians,2.908257,109
1,12 Angry Men,4.344000,125
2,2001: A Space Odyssey,3.969112,259
3,Absolute Power,3.370079,127
4,"Abyss, The",3.589404,151
...,...,...,...
329,Willy Wonka and the Chocolate Factory,3.631902,326
330,"Wizard of Oz, The",4.077236,246
331,"Wrong Trousers, The",4.466102,118
332,Young Frankenstein,3.945000,200


In [68]:
df_top.sort_values(ascending=False, by='Avg_rating').head(25) #display top 25 movies on the basis of average rating

,movie title,Avg_rating,Rating_counts
63,"Close Shave, A",4.491071,112
260,Schindler's List,4.466443,298
331,"Wrong Trousers, The",4.466102,118
54,Casablanca,4.456790,243
269,"Shawshank Redemption, The",4.445230,283
248,Rear Window,4.387560,209
319,"Usual Suspects, The",4.385768,267
291,Star Wars,4.358491,583
1,12 Angry Men,4.344000,125
59,Citizen Kane,4.292929,198


In [69]:
df_top.sort_values(ascending=False, by='Rating_counts').head(25) #display top 25 movies on the basis of rating counts

,movie title,Avg_rating,Rating_counts
291,Star Wars,4.358491,583
69,Contact,3.803536,509
107,Fargo,4.155512,508
251,Return of the Jedi,4.007890,507
178,Liar Liar,3.156701,485
99,"English Patient, The",3.656965,481
261,Scream,3.441423,478
309,Toy Story,3.878319,452
8,Air Force One,3.631090,431
159,Independence Day (ID4),3.438228,429


- *There are 334 movies in total which have received ratings more than 100 times*
- *As per the rating counts, **Star Wars** has received maximum number counts with an average rating of 4.35*
- *There are few movies which have least rating counts like Paris Was a Woman	,Á köldum klaka (Cold Fever)*

### Finding the relationship between demographic of the users and their ratings of the movies

In [70]:
# Merging user dataset with movie and ratings(already merged : df_merge) dataset
df_merge_all = df_merge.merge(user, on = 'user id', how = 'inner')

In [71]:
df_merge_all.head()

,movie id,movie title,release date,Action,Adventure,Animation,Childrens,Comedy,Crime,Documentary,...,War,Western,Number of Genres,user id,rating,timestamp,age,gender,occupation,zip code
0,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,3,308,4,887736532,60,M,retired,95076
1,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,3,287,5,875334088,21,M,salesman,31211
2,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,3,148,4,877019411,33,M,engineer,97006
3,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,3,280,4,891700426,30,F,librarian,22903
4,1,Toy Story,1-Jan-95,0,0,1,1,1,0,0,...,0,0,3,66,3,883601324,23,M,student,80521


**checking on various attributes of the users in order to know the preferences for the movies and their genres**

In [72]:
user['gender'].value_counts(normalize=True) #this will give the proportion of males and females users extracted from User dataset

gender
M    0.710498
F    0.289502
Name: proportion, dtype: float64

In [73]:
df_merge_all.groupby('gender').rating.mean() #grouping the dataset on 'gender' column and taking the average of ratings

gender
F    3.531510
M    3.529333
Name: rating, dtype: float64

- *The average rating given by Males and Females is almost equal with a value of around 3.53*

In [74]:
df_merge_all.groupby('occupation').rating.mean().sort_values(ascending= False) #grouping the dataset on 'occupation' column and taking the average of ratings.

occupation
none             3.777778
lawyer           3.735316
doctor           3.688889
educator         3.670692
artist           3.653380
administrator    3.635646
scientist        3.611273
salesman         3.582944
programmer       3.568205
librarian        3.560781
other            3.552335
engineer         3.541473
technician       3.532097
student          3.515167
marketing        3.485641
retired          3.466750
entertainment    3.440783
writer           3.376152
executive        3.349794
homemaker        3.301003
healthcare       2.896220
Name: rating, dtype: float64

- *Users who are not working have given highest ratings as compared to users who are working professionals*
- *Healthcare workers have given lowest ratings on an average*

<h1><font size=5>Insights</font></h1>


- *Around 71% of users are Males while only 29% users are females*
- *The average rating given by Males and Females is almost equal with a value of around 3.53*
- *Users who are not working have given highest ratings as compared to users who are working professionals*
- *Healthcare workers have given lowest ratings on an average*
- *The low ratings from the healthcare sector is majorly driven by female workers*
- *Only male doctors have given ratings. No female doctors have given ratings as per the observations*

### **Executive Summary**

- *Movies are distributed across 18 genres in total*
- *The top five genres with the highest number of movies are 'Drama', 'Comedy', 'Action', 'Thriller', and 'Romance'*
- *Out of 1680 movies, 50% of the movies have more than one genre, while the rest have a single genre*
- The *'Film-Noir' genre has the highest average ratings with an average rating of 3.92, while the 'Fantasy' genre has the lowest average ratings with an average rating of 3.21*
- *Among 18 genres, 72% of the genres are rated above average 3.5*
- *Movies like **Great Day in Harlem, A**, **Prefontaine**, etc. are top-rated movies, while movies like **Shadow of Angels (Schatten der Engel)**, **Power 98**, etc. are least rated*
- *There are 334 movies in total which have been rated more than 100 times*
- *The movie **Close Shave, A** has the highest average rating and was rated 112 times, while the **Star Wars** movie received the highest number of ratings with an average rating of 4.35*
- *Around 71% of users are Males while only 29% users are females*
- *The average rating given by Males and Females is almost equal with a value of around 3.53*
- *Users who are not working have rated movies higher compared to users who are working professionals*
- *Healthcare workers have given the lowest ratings on an average*

<font size=5 color="Yellow">**Dennis Umoh**</font>

<font size=3 color="Blue">Data Scientist/ML Engineer</font>

___